Microsoft.Toolkit.Uwp.UI.Controls.Layout/WrapLayout/WrapLayout.cs

```diff
@@ -254,10 +254,18 @@ protected override Size MeasureOverride(VirtualizingLayoutContext context, Size
             // for the last condition it is zeros so adding it will make no difference
             // this way is faster than an if condition in every loop for checking the last item
             totalMeasure.U = parentMeasure.U;
+
+            // Propagating an infinite size causes a crash. This can happen if the parent is scrollable and infinite in the opposite
+            // axis to the pannel. Clearing to zero prevents the crash.
+            // This is likely an incorrect use of the control by the developer, however we need stability here so setting a default that wont crash.
+            if (double.IsInfinity(totalMeasure.U))
+            {
+                totalMeasure.U = 0.0;
```

Comment:
If the U is zero, then won't that completely mess up layout for other items that are within the same panel? Example: WrapLayout in a Canvas

Reply:
@skendrot the Canvas also doesn't give it finite bounds in that case (I tested it both in and out of the ScrollViewer). From what we were seeing, this was really a case where the container is unbounded in **both** directions, this was causing only a single line of elements to appear (with the fix, otherwise it was crashing in all cases).

In this use case the control probably isn't being used correctly by the developer as then you'd just want to use a StackPanel.

We just didn't want the control itself to crash in this case, as it was.